src/System.Runtime/tests/Performance/Perf.String.cs

```diff
@@ -232,28 +233,104 @@ public void Substring_int_int(int size)
                         testString.Substring(startIndex, length);
         }
 
-        [Benchmark]
-        [MemberData(nameof(TestStringSizes))]
-        public void ToLower(int size)
+        [Flags]
+        public enum ChangeCaseOptions
         {
-            PerfUtils utils = new PerfUtils();
-            string testString = utils.CreateString(size);
+            None = 0,
+            UniqueString = 0x1,
+
+            MiddleTurkishI = 0x2,
+            MiddleDifferentCase = 0x4,
+            AllDifferentCase = 0x8,
+        }
+
+        public static IEnumerable<object[]> ChangeCaseMemberData() =>
+            from size in new[] { 1, 10, 500 }
+            from culture in new[] { "en-US" }
+            from options in new[]
+            {
+                ChangeCaseOptions.None,
+                ChangeCaseOptions.MiddleTurkishI,
```

Comment:
>MiddleTurkishI [](start = 34, length = 14)

I am not getting this, why we have this option while not using the Turkish culture? it would make sense if you add the Turkish culture there.

Reply:
Because if we encounter a non-ASCII character, it knocks us off the ASCII path, regardless of culture.

Reply:
I am seeing you are creating the whole string as ascii. right? maybe I am missing something here.

Reply:
are you saying you are trying to test the fast path even when having Turkish I?

Reply:
> I am seeing you are creating the whole string as ascii. right? maybe I am missing something here.

When this option is set, it's creating the whole string as ASCII except for the middle character, which ends up being set to a non-ASCII character (\u0131).  Thus when the code gets to that character, it'll be knocked off the ASCII path and will be forced to fall back to doing the OS-based culture-aware comparison.

Reply:
Yes, I understand this, but still this has nothing to do with Turkish.  this is my point. it would be better either to add Turkish culture to test this Turkish I, or you change the name of this enum value to be something like NonAscii.

Reply:
> Yes, I understand this, but still this has nothing to do with Turkish. this is my point. it would be better either to add Turkish culture to test this Turkish I, or you change the name of this enum value to be something like NonAscii.

As long as I was choosing a non-ASCII character, I chose one that made it easy to then locally add "tr-TR" as another culture and see what, if any, impact there was.

I don't feel strongly about it, though.  This has already been merged; you're welcome to submit a PR to change the name of the enum value if you want.